tools/compare.go

```diff
@@ -15,125 +15,291 @@
 package tools
 
 import (
-	"encoding/json"
+	"context"
 	"fmt"
-	"io"
-	"strconv"
-	"sync"
-
-	"github.com/google/go-cmp/cmp"
 	"github.com/spf13/cobra"
 	"github.com/streamingfast/bstream"
+	"github.com/streamingfast/cli"
 	"github.com/streamingfast/dstore"
+	"github.com/streamingfast/eth-go/rpc"
 	pbeth "github.com/streamingfast/firehose-ethereum/types/pb/sf/ethereum/type/v2"
+	"github.com/streamingfast/substreams/block"
+	"go.uber.org/multierr"
+	"google.golang.org/protobuf/proto"
+	"io"
+	"math"
+	"os"
+	"os/exec"
+	"strconv"
+	"sync"
 )
 
 var compareBlocksCmd = &cobra.Command{
-	Use:   "compareblocks <blocks_store_a> <blocks_store_b> <prefix>",
-	Short: "Checks for any differences between merge files of two different stores. (To compare the output of two different instrumentations, for example)",
-	Args:  cobra.ExactArgs(3),
-	RunE:  compareBlocksE,
+	Use:   "compare-blocks <expected_bundle> <actual_bundle> [<block_range>]",
+	Short: "Checks for any differences between two block stores between a specified range. (To compare the likeness of two block ranges, for example)",
+	Long: cli.Dedent(`
+		compare-blocks takes in two paths to stores of merged blocks and a range specifying the blocks you want to compare (written as: start-finish).
+		It will output the status of the likeness of every million blocks, on completion, or on encountering a difference. 
+		Increments that contain a difference will be communicated as well as the blocks within that contain differences.
+		Increments that do not have any differences will be outputted as identical.
+		
+		After passing through the blocks, it will output instructions on how to locate a specific difference based on the
+		blocks that were given. This is done by applying the --diff flag before your args. 
+
+		Commands inputted with --diff will display the blocks that have differences, as well as the difference. 
+	`),
+	Args: cobra.ExactArgs(3),
+	RunE: compareBlocksE,
+	Example: cli.Dedent(`
+		# Run over full block range
+		fireeth tools compare-blocks sf_bundle/ cs_bundle/ 0-16000000
+
+		# Run over specific block range, displaying differences in blocks
+		fireeth tools compare-blocks --diff sf_bundle/ cs_bundle/ 100-200
```

Comment:
Let's remove `sf` and `cs` and change to `expected` `actual` so `expected_bundles`, `actual_bundles`